lego/apps/files/views.py

```diff
@@ -46,6 +46,21 @@ def create(self, request, *args, **kwargs):
             status=status.HTTP_201_CREATED,
         )
 
+    @decorators.action(
+        detail=True,
+        methods=["PATCH"],
+        permission_classes=[permissions.AllowAny],
+    )
```

Comment:
This definitely needs other permissions :))

I realize this might actually be a little tricky, but here are my thoughts:
- Have permission class set to authenticated f.ex.
- Do a permission check on the object. Since I don't believe we have a permission class for `File`s?? you could just do the permission check inline here in the view (or create a one-off function for it). I think you'd need to get the related object of the file (`Article`, `Event` (maybe just do `Content` actually)) and do a permission check on that object.

The _easier_ way would be to not allow this patch at all and just set this flag on creating the event/article. What is actually the reasoning for having this endpoint?

Reply:
When I read some more of the `FileField` code, I see that much of this _might_ be covered there? Could you confirm that? If that's the case you could also use the logic that exists there and apply it here. You could also just use the `FileField` directly to simplify things I guess, even though that might be a bit weird.

If you could write up some tests to back up any hypothesis/claims on how the permission checks work here that would be fantastic

Reply:
Added EDIT event permission check